### ISSUE TITLE
get "DeprecationWarning: There is no current event loop" warning

### ISSUE THREAD

**Issue by jcyrss** [zkmkarlsruhe/baton#1]
Run this script in Python3.10, we will get "DeprecationWarning: There is no current event loop" warning.

That's caused by 

```py
loop = asyncio.get_event_loop()
```

Suggest to use this

```py
loop = asyncio.new_event_loop()
asyncio.set_event_loop(loop)
```

<br>

And, here

```py
class UDPReceiver:

    @staticmethod
    def create(loop, local_addr, server, verbose):
        task = asyncio.Task(loop.create_datagram_endpoint(
            lambda: UDPReceiver(server=server, verbose=verbose),
            local_addr=local_addr, allow_broadcast=True))
        _, udpreceiver = loop.run_until_complete(task)
        return udpreceiver
```

It creates a Task instance without specifying event loop, that will cause it to use current running loop which does not exists at that time. It will also bring "DeprecationWarning: There is no current event loop" warning.

Suggest to change to this

```py
class UDPReceiver:

    @staticmethod
    def create(loop, local_addr, server, verbose):
        coro = loop.create_datagram_endpoint(
            lambda: UDPReceiver(server=server, verbose=verbose),
            local_addr=local_addr, allow_broadcast=True)

        task = asyncio.Task(coro,loop=loop)

        _, udpreceiver = loop.run_until_complete(task)
        return udpreceiver
```

The same modification should apply to `UDPSender`

**Comment by danomatika** [comment]
Thanks for the report. I don't see warnings on my system with Python 3.11.3. I will see if there is a workaround.

**Comment by danomatika** [comment]
Ok, in reading the Python 3.11 [changelog](https://docs.python.org/3/whatsnew/changelog.html#python-3-11-0-final), I confirm the issue is really down to deprecating the behavior of `asynicio.get_event_loop()` towards Python 3.12 - 3.14: https://github.com/python/cpython/issues/93453.

Your proposed fix of explicitly creating the event loop before using it should be fine. I don't think anything else is needed since the class instances all start their tasks using the loop in their respective `create` methods.

**Comment by danomatika** [comment]
Latest commit creates the event loop explicitly. Can you test on your system?

**Comment by jcyrss** [comment]
> Latest commit creates the event loop explicitly. Can you test on your system?

Yeah, sure. But I don't see any changes to the code file 'baton.py'

![image](https://github.com/zkmkarlsruhe/baton/assets/10496014/72050f36-3849-48d1-bb81-573d459d9de0)

**Comment by danomatika** [comment]
Oops. I pushed to our internal Gitlab server. The commit is now pushed here.
